app/app/explore/pools/[asset]/PoolDetail.tsx

```diff
@@ -8,18 +8,64 @@ import {
 } from "@/app/utils";
 import { PoolDetail as IPoolDetail } from "@/midgard";
 import { BackArrow } from "@shared/components/svg";
+import AddLiquidityModal from "@/app/explore/components/AddLiquidityModal";
+import { useState, useEffect } from "react";
 import { TopCard } from "@/app/components/TopCard";
+import { useLiquidityPosition } from "@/hooks/useLiquidityPosition";
+import { useAppState } from "@/utils/context";
 
 interface PoolDetailProps {
   pool: IPoolDetail;
   runePriceUSD: number;
 }
 
 export default function PoolDetail({ pool, runePriceUSD }: PoolDetailProps) {
+  const { wallet } = useAppState();
+  const [showAddLiquidityModal, setShowAddLiquidityModal] = useState(false);
+  const { position, loading, error, getMemberDetails, removeLiquidity } =
+    useLiquidityPosition({ pool });
+
   // Calculate pool metrics
   const formattedTVL = getFormattedPoolTVL(pool, runePriceUSD);
   const volumeDepthRatio = calculateVolumeDepthRatio(pool, runePriceUSD);
 
+  // Fetch position when wallet connects
+  useEffect(() => {
+    if (wallet?.address) {
+      getMemberDetails(wallet.address, pool.asset);
+    }
+  }, [wallet?.address, pool.asset, getMemberDetails]);
+
+  const handleRemove = async () => {
+    if (!wallet?.address) {
+      // Show connect wallet modal or error
+      return;
+    }
+
+    try {
+      await removeLiquidity({
+        asset: pool.asset,
+        percentage: 100, // Remove all - could make this configurable
+        address: wallet.address,
+      });
+    } catch (err) {
+      console.error("Failed to remove liquidity:", err);
+      // Handle error - could show toast/notification
+    }
+  };
+
+  // Calculate position metrics
+  const assetValue = position ? parseFloat(position.assetDeposit) / 1e8 : 0;
```

Comment:
don't we also need to get the rune deposit value in order to get the total deposit value?

---

We ideally want to display Principle + Yield, but we can't yet, because TC doesn't separate (yet) For now we can compute Deposit and Gain (but GAIN can be negative unfortunately) In Q1, we will hopefully fix this - see TC gitlab

 We can't compute "yield" yet, since yield on TC is not separated from principle (it is co-mingled), so let's call it "Gain" for now Deposit: display as L1 ($USD) eg 1.2 BTC ($85,234.40)

```
https://thornode.ninerealms.com/thorchain/pool/AVAX.AVAX/liquidity_providers
R0: rune_deposit_value: "4922525291", // compute to asset value
A0: asset_deposit_value: "1500000000" // add

total_deposit = R0 + A0
```
 Gain: display as L1 ($USD) 
```
https://thornode.ninerealms.com/thorchain/pool/AVAX.AVAX
https://thornode.ninerealms.com/thorchain/pool/AVAX.AVAX/liquidity_providers

R1: (units / LP_units) * balance_rune // compute to asset value
A1: (units / LP_units) * balance_asset

total_deposit_current = R1 + A1
Gain = total_deposit_current - total_deposit
```
 When we roll out yield-separation on TC, we can accurately track yield and will deprecate the "gain" computation.